Daemon did not reply after PR #35 merge (log excerpt attached)

2026-03-04 17:05:49,859 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:06:01,021 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:06:12,184 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:06:23,344 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:06:34,451 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:06:45,565 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:06:56,726 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:07:07,889 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:07:19,049 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:07:30,162 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:07:41,272 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:07:52,433 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:08:03,594 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:08:14,755 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:08:25,918 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:08:37,078 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:08:48,241 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:08:59,404 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:09:10,565 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:09:21,673 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:09:32,887 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:09:44,049 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:09:55,211 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:10:06,374 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:10:17,477 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:10:28,697 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:10:39,859 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:10:50,968 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:11:02,180 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:11:13,341 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:11:24,446 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:11:35,562 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:11:46,724 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:11:57,886 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:12:09,048 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:12:20,210 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:12:31,371 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:12:42,533 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:12:53,695 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:13:04,856 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:13:15,963 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:13:27,076 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:13:38,339 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:13:49,502 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:14:00,664 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:14:11,771 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:14:22,987 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:14:34,155 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:14:35,736 [INFO] corphish.daemon: Daemon started, listening on chat 8757952223
2026-03-04 17:14:36,609 [INFO] httpx: HTTP Request: POST https://api.telegram.org/bot8645862118:AAETB7EJPgjPc2GdSq4gPKvH17QqiUFvt68/getUpdates "HTTP/1.1 200 OK"
2026-03-04 17:14:36,613 [INFO] corphish.daemon: [user] Hello the fourth
2026-03-04 17:14:36,614 [INFO] claude_agent_sdk._internal.transport.subprocess_cli: Using bundled Claude Code CLI: /Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/claude_agent_sdk/_bundled/claude
2026-03-04 17:14:40,881 [INFO] corphish.daemon: [assistant] Hello. What do you need?
2026-03-04 17:14:40,885 [ERROR] asyncio: unhandled exception during asyncio.run() shutdown
task: <Task finished name='Task-5' coro=<<async_generator_athrow without __name__>()> exception=RuntimeError('Attempted to exit cancel scope in a different task than it was entered in')>
Traceback (most recent call last):
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/claude_agent_sdk/_internal/client.py", line 143, in process_query
    yield message
GeneratorExit

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/claude_agent_sdk/_internal/client.py", line 146, in process_query
    await query.close()
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/claude_agent_sdk/_internal/query.py", line 622, in close
    await self._tg.__aexit__(None, None, None)
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/anyio/_backends/_asyncio.py", line 789, in __aexit__
    if self.cancel_scope.__exit__(type(exc), exc, exc.__traceback__):
       ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/anyio/_backends/_asyncio.py", line 461, in __exit__
    raise RuntimeError(
    ...<2 lines>...
    )
RuntimeError: Attempted to exit cancel scope in a different task than it was entered in
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/zhlhan/.local/share/corphish/corphish/__main__.py", line 33, in <module>
    main()
    ~~~~^^
  File "/Users/zhlhan/.local/share/corphish/corphish/__main__.py", line 29, in main
    asyncio.run(_async_main())
    ~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.14/3.14.3_1/Frameworks/Python.framework/Versions/3.14/lib/python3.14/asyncio/runners.py", line 204, in run
    return runner.run(main)
           ~~~~~~~~~~^^^^^^
  File "/opt/homebrew/Cellar/python@3.14/3.14.3_1/Frameworks/Python.framework/Versions/3.14/lib/python3.14/asyncio/runners.py", line 127, in run
    return self._loop.run_until_complete(task)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "/opt/homebrew/Cellar/python@3.14/3.14.3_1/Frameworks/Python.framework/Versions/3.14/lib/python3.14/asyncio/base_events.py", line 719, in run_until_complete
    return future.result()
           ~~~~~~~~~~~~~^^
  File "/Users/zhlhan/.local/share/corphish/corphish/__main__.py", line 24, in _async_main
    await dispatch(args)
  File "/Users/zhlhan/.local/share/corphish/corphish/cli.py", line 210, in dispatch
    await run_daemon()
  File "/Users/zhlhan/.local/share/corphish/corphish/daemon.py", line 110, in run_daemon
    await send_message_fn(bot, chat_id, reply)
  File "/Users/zhlhan/.local/share/corphish/corphish/chat.py", line 54, in send_message
    return await bot.send_message(chat_id=chat_id, text=text)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/telegram/_bot.py", line 1122, in send_message
    return await self._send_message(
           ^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<21 lines>...
    )
    ^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/telegram/_bot.py", line 819, in _send_message
    result = await self._post(
             ^^^^^^^^^^^^^^^^^
    ...<7 lines>...
    )
    ^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/telegram/_bot.py", line 703, in _post
    return await self._do_post(
           ^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
    )
    ^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/telegram/_bot.py", line 732, in _do_post
    result = await request.post(
             ^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
    )
    ^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/telegram/request/_baserequest.py", line 198, in post
    result = await self._request_wrapper(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<7 lines>...
    )
    ^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/telegram/request/_baserequest.py", line 305, in _request_wrapper
    code, payload = await self.do_request(
                    ^^^^^^^^^^^^^^^^^^^^^^
    ...<7 lines>...
    )
    ^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/telegram/request/_httpxrequest.py", line 279, in do_request
    res = await self._client.request(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
    )
    ^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/httpx/_client.py", line 1540, in request
    return await self.send(request, auth=auth, follow_redirects=follow_redirects)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/httpx/_client.py", line 1629, in send
    response = await self._send_handling_auth(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<4 lines>...
    )
    ^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/httpx/_client.py", line 1657, in _send_handling_auth
    response = await self._send_handling_redirects(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<3 lines>...
    )
    ^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/httpx/_client.py", line 1694, in _send_handling_redirects
    response = await self._send_single_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/httpx/_client.py", line 1730, in _send_single_request
    response = await transport.handle_async_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/httpx/_transports/default.py", line 394, in handle_async_request
    resp = await self._pool.handle_async_request(req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/zhlhan/.local/share/corphish/.venv/lib/python3.14/site-packages/httpcore/_async/connection_pool.py", line 256, in handle_async_request
    raise exc from None

## Log Diagnosis

### Timeline of events

| Time | Event |
|------|-------|
| 17:05 – 17:14:34 | Telegram `getUpdates` polling running normally (no daemon yet) |
| **17:14:35.736** | Daemon starts, logs "listening on chat 8757952223" |
| **17:14:36.613** | User message "Hello the fourth" received and logged |
| **17:14:40.881** | Claude replies "Hello. What do you need?" — **logged successfully** (daemon.py:107) |
| **17:14:40.885** | **Two exceptions fire back-to-back, daemon crashes** |

### What failed

There are **two distinct errors** in the log, and the first one causes the second:

#### Error 1 — SDK async generator cleanup (the root cause)

```
[ERROR] asyncio: unhandled exception during asyncio.run() shutdown
RuntimeError: Attempted to exit cancel scope in a different task than it was entered in
```

After `client.send()` returns (via `break` at claude_client.py:141), the SDK's `query()` async generator is left open. Python's GC eventually finalises it, which triggers `GeneratorExit` → `query.close()` → anyio cancel-scope teardown. But the teardown runs in a **different asyncio Task** than the one that opened the scope, so anyio raises `RuntimeError`.

This is the **same cancel-scope cross-task issue** PR #35 tried to fix. The `break` approach avoids *explicit* `aclosing()` but still leaves the generator to be cleaned up implicitly — and that implicit cleanup hits the same anyio bug.

#### Error 2 — Telegram `send_message` network failure (the consequence)

```
File "daemon.py", line 110, in run_daemon
    await send_message_fn(bot, chat_id, reply)
...
httpcore._async.connection_pool ... raise exc from None
```

The traceback is truncated, but the chain is clear: the SDK cleanup RuntimeError poisons the asyncio event loop during its shutdown phase, which interrupts the in-flight `send_message` HTTP request. The httpx/httpcore connection pool raises a network-level error because the loop is being torn down underneath it.

**Key observation**: daemon.py:109-115 has a `try/except Exception` around `send_message`, but it never fires — the process is already crashing from the asyncio-level error, not from a normal exception within the `send_message` coroutine.

### Root cause

PR #35's `break` approach doesn't fully solve the problem. The `break` exits the `async for` loop cleanly, but the underlying async generator from `sdk.query()` is **not explicitly closed**. Python will GC-finalize it later (often during `asyncio.run()` shutdown), which triggers the cross-task cancel-scope RuntimeError that takes down the entire process — including the Telegram send that was in-flight or about to execute.

### Fix plan

**1. Explicitly drain or close the SDK generator after `break` (claude_client.py)**

After `break`, explicitly close the generator in the same task context to prevent deferred GC cleanup:

```python
async def send(self, user_text: str) -> str:
    last_text = ""
    result_text = None
    gen = self._query(prompt=user_text, options=self._options)

    try:
        async for message in gen:
            if isinstance(message, AssistantMessage):
                parts = [
                    block.text
                    for block in message.content
                    if isinstance(block, TextBlock)
                ]
                if parts:
                    last_text = "\n".join(parts)
            elif isinstance(message, ResultMessage):
                if message.result:
                    result_text = message.result
                    break
    finally:
        # Close the generator in the SAME task that opened it,
        # preventing the cross-task cancel-scope RuntimeError.
        try:
            await gen.aclose()
        except RuntimeError:
            pass  # Already closed or cancel-scope mismatch — safe to ignore

    return result_text or last_text
```

The `try/finally` + `await gen.aclose()` ensures cleanup happens in the correct task context. The inner `except RuntimeError` is a safety net in case the SDK still misbehaves during explicit close.

**2. Add structured logging around the send path (daemon.py)**

Currently the only visibility into the send outcome is the generic `logger.exception`. Add explicit success/timing logs:

```python
logger.info("[assistant] %s", reply)

try:
    await send_message_fn(bot, chat_id, reply)
    logger.info("Telegram reply delivered for message: %s", user_text)
except Exception:
    logger.exception(
        "Failed to send reply via Telegram for message: %s",
        user_text,
    )
```

**3. Guard against event-loop-level crashes during send (daemon.py)**

Wrap the entire per-message processing block in an additional safety net so that an unrelated asyncio error (like SDK cleanup) doesn't take down the daemon:

```python
except BaseException:
    logger.exception("Unexpected fatal error processing message: %s", user_text)
    # Don't re-raise — keep the daemon loop alive
```

### Summary

| Layer | Problem | Fix |
|-------|---------|-----|
| SDK generator | Implicit GC cleanup in wrong task → RuntimeError | Explicit `await gen.aclose()` in `finally` with RuntimeError guard |
| Telegram send | Killed by asyncio shutdown cascade | Structured success/failure logging to distinguish send errors from loop crashes |
| Daemon loop | No defence against BaseException from asyncio internals | Catch BaseException around per-message block to keep loop alive |

--- a/corphish/claude_client.py
+++ b/corphish/claude_client.py
@@ -123,21 +123,28 @@ class ClaudeClient:
         """
         last_text = ""
         result_text = None
+        gen = self._query(prompt=user_text, options=self._options)
 
-        async for message in self._query(
-            prompt=user_text, options=self._options
-        ):
-            if isinstance(message, AssistantMessage):
-                parts = [
-                    block.text
-                    for block in message.content
-                    if isinstance(block, TextBlock)
-                ]
-                if parts:
-                    last_text = "\n".join(parts)
-            elif isinstance(message, ResultMessage):
-                if message.result:
-                    result_text = message.result
-                    break
+        try:
+            async for message in gen:
+                if isinstance(message, AssistantMessage):
+                    parts = [
+                        block.text
+                        for block in message.content
+                        if isinstance(block, TextBlock)
+                    ]
+                    if parts:
+                        last_text = "\n".join(parts)
+                elif isinstance(message, ResultMessage):
+                    if message.result:
+                        result_text = message.result
+                        break
+        finally:
+            await gen.aclose()
+            # Yield control so any pending anyio cancel-scope cleanup from the
+            # SDK's internal task group settles before the caller does further
+            # I/O.  Without this, a leaked cancellation can fire on the next
+            # await (e.g. send_message), causing CancelledError.  See #33.
+            await asyncio.sleep(0)
 
         return result_text or last_text

--- a/tests/test_claude_client.py
+++ b/tests/test_claude_client.py
@@ -388,11 +388,11 @@ async def test_lock_serialises_calls():
 
 
 async def test_send_closes_generator_on_early_return():
-    """send() must stop consuming after ResultMessage.result via break.
+    """send() must close the generator in the same task after breaking.
 
-    Regression test for issue #27: the generator is no longer wrapped in
-    aclosing(); instead we break out of the loop. The result must still be
-    correct and no messages after the break should affect the return value.
+    Regression test for issues #27 and #36: the generator is explicitly closed
+    via aclose() in the finally block so the SDK's cancel scope is torn down in
+    the originating task, not deferred to GC in a different task.
     """
     from claude_agent_sdk import AssistantMessage, TextBlock, ResultMessage
 
@@ -417,15 +417,13 @@ async def test_send_closes_generator_on_early_return():
             yield AssistantMessage(
                 content=[TextBlock(text="should not reach")], model="test"
             )
-        except GeneratorExit:
-            closed = True
-            raise
         finally:
             closed = True
 
     client = _make_client(query_fn=gen_query)
     result = await client.send("test")
     assert result == "final"
+    assert closed, "async generator was not closed after early break"
 
 
 async def test_send_closes_generator_on_normal_exhaustion():

